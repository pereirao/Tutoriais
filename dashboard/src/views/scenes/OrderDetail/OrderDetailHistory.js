// @flow

import React, {PureComponent} from 'react';
import {connect} from 'react-redux';

import I18n from '../../../localization';
import {order_helpers} from '../../../business/order';
import style_sheet from './OrderDetail.module.css';
import OrderTimeline from './OrderTimeline';
import OrderComments from './OrderComments';
import OrderExtras from './OrderExtras';
import {session_selectors} from '../../../business/session';

const {currentSupplier} = session_selectors;

const {
  commentIds
} = order_helpers;

export class OrderDetailHistoryModal extends PureComponent {
  render() {
    const {order, isMinibarClosed, supplier} = this.props;
    let extrasWrapper = (null);
    if (hasExtras(supplier)) {
      extrasWrapper = (
        <div className={style_sheet.extrasWrapper}>
          <div className={style_sheet.sectionHeader}><p
            className={style_sheet.sectionTitle}>{I18n.t('ui.header.order_extras')}</p></div>
          <div className={style_sheet.extrasSectionContents}>
            <OrderExtras order={order}/>
          </div>
        </div>
      );
    }
    return (
      <div className={style_sheet.flexRow}>
        <div className={style_sheet.timelineWrapper}>
          <div className={style_sheet.sectionHeader}><p
            className={style_sheet.sectionTitle}>{I18n.t('ui.header.order_timeline')}</p></div>
          <div className={style_sheet.sectionContents}>
            <OrderTimeline order={order}/>
          </div>
        </div>
        <div className={style_sheet.commentsWrapper}>
          <div className={style_sheet.sectionHeader}><p
            className={style_sheet.sectionTitle}>{I18n.t('ui.header.order_comments')}</p></div>
          <div className={style_sheet.commentSectionContents}>
            <OrderComments order_id={order.id} isMinibarClosed={isMinibarClosed} comment_ids={commentIds(order)}/>
          </div>
        </div>
        {extrasWrapper}
      </div>
    );
  }
}

function hasExtras(supplier) {
  return supplier.partner_config &&
    supplier.partner_config.extrasForm &&
    supplier.partner_config.extrasForm.fields &&
    supplier.partner_config.extrasForm.fields.length > 0;
}

const OrderDetailHistorySTP = state => {
  return ({
    supplier: currentSupplier(state)
  });
};

export const OrderDetailHistory = connect(OrderDetailHistorySTP)(OrderDetailHistoryModal);


export default OrderDetailHistory;
