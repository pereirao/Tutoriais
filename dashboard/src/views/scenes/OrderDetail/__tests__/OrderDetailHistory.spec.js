import React from 'react';
import order_factory from '../../../../business/order/__tests__/order.factory';

// NOTE: the named export isn't a connected component
import OrderDetailHistory from '../OrderDetailHistory';

it('renders correctly', () => {
  expect(shallow(
    <OrderDetailHistory order={order_factory.build()} />
  )).toMatchSnapshot();
});
