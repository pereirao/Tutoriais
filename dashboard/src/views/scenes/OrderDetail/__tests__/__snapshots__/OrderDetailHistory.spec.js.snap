// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flexRow"
>
  <div
    className="timelineWrapper"
  >
    <div
      className="sectionHeader"
    >
      <p
        className="sectionTitle"
      >
        ORDER TIMELINE
      </p>
    </div>
    <div
      className="sectionContents"
    >
      <Connect(OrderTimeline)
        order={
          Object {
            "address": Object {
              "address1": "21 Jump Street",
              "address2": "Apt 5",
              "city": "Brooklyn",
              "coords": Object {
                "lat": 40.719507,
                "lng": -73.941461,
              },
              "state": "NY",
              "zip_code": 11206,
            },
            "amounts": Object {
              "delivery_fee": 5,
              "minibar_promos": 2.35,
              "store_discounts": 2,
              "subtotal": 43.47,
              "tax": 3.48,
              "tip": 4.35,
              "total": 51.95,
            },
            "billing": Object {
              "cc_type": "Amex",
              "last_digits": "4321",
            },
            "comment_ids": Array [],
            "customer_name": "Biggie Smalls",
            "delivery_method": Object {
              "maximum_delivery_expectation": 60,
              "type": "on_demand",
            },
            "driver_id": 15,
            "id": 1,
            "notes": "Ring the doorbell",
            "number": 1,
            "order_items": Array [],
            "order_time": "2017-01-01T00:00:00-05:00",
            "receipt_url": "http://www.minibardelivery.com",
            "recipient_info": Object {
              "long_name": "Biggie Smalls - East Coast",
              "phone": "5555555555",
              "short_name": "Biggie Smalls",
            },
            "state": "paid",
            "type_tags": Object {
              "corporate": false,
              "gift": false,
              "out_of_hours": false,
              "scheduled": false,
              "vip": true,
            },
          }
        }
      />
    </div>
  </div>
  <div
    className="commentsWrapper"
  >
    <div
      className="sectionHeader"
    >
      <p
        className="sectionTitle"
      >
        ORDER COMMENTS
      </p>
    </div>
    <div
      className="commentSectionContents"
    >
      <Connect(OrderComments)
        comment_ids={Array []}
        order_id={1}
      />
    </div>
  </div>
</div>
`;
